plugins/ltac2/tac2def.ml

```diff
@@ -0,0 +1,86 @@
+(************************************************************************)
+(*         *   The Coq Proof Assistant / The Coq Development Team       *)
+(*  v      *         Copyright INRIA, CNRS and contributors             *)
+(* <O___,, * (see version control and CREDITS file for authors & dates) *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+open Proofview
+open Tac2ffi
+
+(* Local shortcuts. *)
+type v = Tac2ffi.valexpr
+type 'r of_v = v -> 'r
+type 'r to_v = 'r -> v
+type 'a with_env = Environ.env -> Evd.evar_map -> 'a
+
+let value : valexpr repr = make_repr Fun.id Fun.id
+
+type (_, _, _) spec =
```

Comment:
I guess the 3rd type is in some sense the return type, but it doesn't seem used so should be removed.

Reply:
It is actually load-bearing. Removing it gives:
```
File "plugins/ltac2/tac2externals.ml", lines 20-25, characters 0-60:
20 | type (_, _) spec =
21 |   | T : 'r to_v -> (v tactic, 'r tactic) spec
22 |   | V : 'r to_v -> (v tactic, 'r) spec
23 |   | E : 'r to_v -> (v tactic, 'r with_env) spec
24 |   | G : 'r to_v -> (v tactic, Goal.t -> 'r) spec
25 |   | F : 'a of_v * ('t, 'f) spec -> (v -> 't , 'a -> 'f) spec
Error: In this definition, a type variable cannot be deduced
       from the type parameters.
```

Reply:
Actually, I can make it work with another level of GADT.

Reply:
I had never seen that error